### PR TITLE
Transaction-suppot, connection count issue fix

### PR DESCRIPTION
**Pull request by sdkottegoda**

### CODE REVIEW

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java [hunk 1]
```diff
@@ -187,11 +182,12 @@ public void storeMessagePart(List<AndesMessagePart> partList) throws AndesExcept
...
+     *
      * @param preparedStatement Prepared statement for storing message content
-     * @param messagePart message content to be stored
+     * @param messagePart       message content to be stored
      * @throws SQLException
      */
-    private void addContentToBatch(PreparedStatement preparedStatement, AndesMessagePart messagePart) throws SQLException{
+    private void addContentToBatch(PreparedStatement preparedStatement, AndesMessagePart messagePart) throws SQLException {
         preparedStatement.setLong(1, messagePart.getMessageID());
         preparedStatement.setInt(2, messagePart.getOffSet());
```
Comment (AnujaLK) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java]: Offset must be one word.

**Comment by AnujaLK** [comment]
Do we have a related JIRA for this?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java (hunk 1 above)
Comment (sdkottegoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java]: Fixed by https://github.com/wso2/andes/pull/381

**Comment by sdkottegoda** [comment]
Created a jira and mentioned it.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java [hunk 2]
```diff
@@ -926,8 +922,8 @@ public void deleteMessageMetadataFromQueue(final String storageQueueName,
         } catch (SQLNonTransientConnectionException sqlConEx) {
             rollback(connection, RDBMSConstants.TASK_DELETING_METADATA_FROM_QUEUE + storageQueueName);
             throw new AndesStoreUnavailableException("error occurred while deleting message metadata from queue " + storageQueueName,
-            sqlConEx.getSQLState(), sqlConEx);
-            
+                    sqlConEx.getSQLState(), sqlConEx);
```
Comment (malakasilva) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java]: Extra line spacing

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java [hunk 3]
```diff
@@ -1984,25 +1971,27 @@ public void rollback() throws AndesException {
...
             }
-            txConnectionsCount.incrementAndGet();
-            Connection con = datasource.getConnection();
+            Connection con = rdbmsConnection.getDataSource().getConnection();
             con.setAutoCommit(false);
             storeMetadataPS = con.prepareStatement(PS_INSERT_METADATA);
             storeContentPS = con.prepareStatement(PS_INSERT_MESSAGE_PART);
             connectionClosed = false;
+            txConnectionsCount.incrementAndGet();
+
```
Comment (malakasilva) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java]: Extra line not required

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java (hunk 2 above)
Comment (sdkottegoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java]: Implementation has changed. The line is not present anymore.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java (hunk 3 above)
Comment (sdkottegoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java]: The implementation has changed and the method is removed in the current code.

**Pull request merged**